pytest-selenium/tests/test_navigation.py

```diff
@@ -0,0 +1,59 @@
+import pytest
+from pages.content import Content
+from tests import markers
+
+
+@markers.test_case("C477321")
+@markers.parametrize("page_slug", [("preface")])
+@markers.nondestructive
+def test_previous_link_hidden_on_first_page(selenium, base_url, book_slug, page_slug):
+
+    # GIVEN: The page is loaded
+    content = Content(selenium, base_url, book_slug=book_slug, page_slug=page_slug).open()
+    toolbar = content.toolbar
+    toc = content.sidebar.toc
+
+    # confirm first page is selected
+    if toc.active_section.get_attribute("textContent") != toc.sections[0].section_title:
+        while content.is_mobile:
+            toolbar.click_toc_toggle_button()
+            break
+        toc.sections[0].click()
+
+    # THEN: The "previous" link should be hidden
+    with pytest.raises(Exception) as exc_info:
+        assert not content.previous_link.is_displayed
```

Comment:
this `with pytest.raises(Exception)` feels like it should be in a helper, you have other helpers for similar things

Reply:
@TomWoodward 
Mike suggested that Assertions has to be within the test and not in the helper functions (as per pytest guidelines)

I still have the other test where assertions are in the helper because, it was checking the same assertion for 3 different elements and did not make sense to me to have 4 lines of code repeated in a test 3 times.
 
This assertion is checking a different exception just one time, so I thought I can comply to the rules of pytest :-). I dont mind doing it the other way though

Reply:
the whole process of catching the exception doesn't have to be directly linked to the assertion. if you did the catch within the `is_displayed` function and returned `false`, then the assertion just becomes `assert not content.previous_link.is_displayed` and all the logic is encapsulated in that function

Reply:
@TomWoodward May be I am not getting your point, but when I tried 
`assert not content.previous_link.is_displayed` fails with `nosuchelement` exception error even before calling the `is_displayed` function because the previous_link is not visible.

I also tried to call  `content.assert_element_not_displayed(content.previous_link)` where
    """def assert_element_not_displayed(self, element):
        with pytest.raises(Exception) as exc_info:
            assert not element.is_displayed
        exception_raised = exc_info.type
        assert "NoSuchElementException" in str(exception_raised)
        return False"""
which would again fail even before calling the helper, because the the argument (previous_link) that we are passing is not visible in the first place.

Reply:
@TomWoodward Ignore my previous comment. I got what you are saying and have updated the test